transports/tcp/src/lib.rs

```diff
@@ -716,6 +636,20 @@ where
                 .unregister(self.listen_addr.ip(), self.listen_addr.port()),
         }
     }
+
+    /// Close the listener.
+    ///
+    /// This will create a [`TransportEvent::ListenerClosed`] event
+    /// and terminate the stream once all remaining events in queue have
+    /// been reported.
+    fn close(&mut self, reason: Result<(), io::Error>) {
+        self.queued_events
+            .push_back(TransportEvent::ListenerClosed {
+                listener_id: self.listener_id,
+                reason,
+            });
+        self.is_closed = true;
```

Comment:
In case the waker is set, you need to call it here to make sure you are getting polled.